### PR TITLE
core,okhttp: add TransportTracer to okhttpclient

### PR DESCRIPTION
**Pull request by zpencer**

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/AbstractClientStream.java [hunk 1]
```diff
@@ -102,11 +101,11 @@ void writeFrame(
   protected AbstractClientStream(
       WritableBufferAllocator bufferAllocator,
       StatsTraceContext statsTraceCtx,
-      @Nullable TransportTracer transportTracer,
+      TransportTracer transportTracer,
       Metadata headers,
       boolean useGet) {
     Preconditions.checkNotNull(headers, "headers");
-    this.transportTracer = transportTracer;
+    this.transportTracer = Preconditions.checkNotNull(transportTracer, "transportTracer");
```
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/internal/AbstractClientStream.java]: optional: statically importing this makes the code shorter usually.

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java [hunk 2]
```diff
@@ -127,6 +128,13 @@ private OkHttpChannelBuilder(String target) {
     }
   }
 
+  @VisibleForTesting
+  final OkHttpChannelBuilder setTransportTracerFactory(
+      TransportTracer.Factory transportTracerFactory) {
```
Comment (carl-mastrangelo) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java]: fits on th prev line?

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java [hunk 3]
```diff
@@ -246,6 +252,23 @@
     this.proxyPassword = null;
     this.tooManyPingsRunnable =
         Preconditions.checkNotNull(tooManyPingsRunnable, "tooManyPingsRunnable");
+    this.transportTracer = Preconditions.checkNotNull(transportTracer);
```
Comment (carl-mastrangelo) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java]: name this param?

File: core/src/main/java/io/grpc/internal/AbstractClientStream.java (hunk 1 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/internal/AbstractClientStream.java]: done

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java (hunk 2 above)
Comment (zpencer) [reply okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java]: too long by just a bit unfortunately

**Pull request merged**